src/plots/geo/geo.js

```diff
@@ -136,21 +135,20 @@ proto.plot = function(geoData, fullLayout, promises) {
 };
 
 proto.onceTopojsonIsLoaded = function(geoData, geoLayout) {
-    var scattergeoData = [],
-        choroplethData = [];
+    var traceData = {};
 
     this.drawLayout(geoLayout);
 
     for(var i = 0; i < geoData.length; i++) {
         var trace = geoData[i];
-        var traceType = trace.type;
 
-        if(traceType === 'scattergeo') scattergeoData.push(trace);
-        else if(traceType === 'choropleth') choroplethData.push(trace);
+        traceData[trace.type] = traceData[trace.type] || [];
+        traceData[trace.type].push(trace);
```

Comment:
and by the way, I should add (cc @cldougl ): 

The reason why geo and cartesian split the data by trace type than call `_module.plot` with all traces of that trace type is to have only one svg layer per trace type per subplot. 

This pattern makes it trivial to use `d3`'s [data/enter/exit](http://bost.ocks.org/mike/join/).